drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/editor/ScenarioSimulationEditorPresenter.java

```diff
@@ -417,6 +414,48 @@ public boolean isDirty() {
         }
     }
 
+    public ScenarioGridWidget getBackgroundGridWidget() {
+        return scenarioBackgroundGridWidget;
+    }
+
+    public void onEditTabSelected() {
+        ScenarioGridWidget scenarioGridWidget = getView().getScenarioGridWidget();
+        setItemMenuForMainGridEnabled(true);
+        scenarioGridWidget.clearSelections();
+        scenarioGridWidget.select();
+        getBackgroundGridWidget().deselect();
+        populateRightDocks(TestToolsPresenter.IDENTIFIER);
+    }
+
+    public void onBackgroundTabSelected() {
+        ScenarioGridWidget backgroundWidget = getBackgroundGridWidget();
+        setItemMenuForMainGridEnabled(false);
+        backgroundWidget.clearSelections();
+        backgroundWidget.select();
+        getView().getScenarioGridWidget().deselect();
+        populateRightDocks(TestToolsPresenter.IDENTIFIER);
+    }
+
+    public void onOverviewSelected() {
+        ScenarioGridWidget scenarioGridWidget = getView().getScenarioGridWidget();
+        ScenarioGridWidget backgroundWidget = getBackgroundGridWidget();
+        setItemMenuForMainGridEnabled(false);
+        scenarioGridWidget.clearSelections();
+        scenarioGridWidget.deselect();
+        backgroundWidget.clearSelections();
+        backgroundWidget.deselect();
+    }
+
+    public void onImportsTabSelected() {
+        ScenarioGridWidget scenarioGridWidget = getView().getScenarioGridWidget();
+        ScenarioGridWidget backgroundWidget = getBackgroundGridWidget();
+        setItemMenuForMainGridEnabled(false);
+        scenarioGridWidget.clearSelections();
+        scenarioGridWidget.deselect();
+        backgroundWidget.clearSelections();
+        backgroundWidget.deselect();
+    }
```

Comment:
Just question, why in case of `scenarioGridWidget` we access via `getView().getScenarioGridWidget()`  and in case of `backgroundWidget` we access via `getBackgroundGridWidget()` what is just getter of `scenarioBackgroundGridWidget`. Looking on class fields we have also `scenarioMainGridWidget` so my question if we need to use `getView().getScenarioGridWidget()` instead of accessing `scenarioMainGridWidget`?

Reply:
@jomarko 
Good catch - thx! All those getter invocations absolutely unneeded